OpenRA.Game/FieldLoader.cs

```diff
@@ -441,6 +442,20 @@ public static object GetValue(string fieldName, Type fieldType, string value, Me
 					addMethod.Invoke(set, new[] { GetValue(fieldName, fieldType.GetGenericArguments()[0], parts[i].Trim(), field) });
 				return set;
 			}
+			else if (fieldType.IsGenericType && fieldType.GetGenericTypeDefinition() == typeof(Dictionary<,>))
+			{
+				var dict = Activator.CreateInstance(fieldType);
+				var addMethod = fieldType.GetMethod("Add", fieldType.GetGenericArguments());
```

Comment:
You call `fieldType.GetGenericArguments()` `2 * <number of children> + 1` times.